File: website/app/Http/Swagger/swagger_models/pasien.php
<?php

/**
  * @OA\Get(
  *      path="/v1/entities/pasien",
  *      operationId="browsePasien",
  *      tags={"pasien"},
  *      summary="Browse Pasien",
  *      description="Returns list of Pasien",
  *      @OA\Response(response=200, description="Successful operation"),
  *      @OA\Response(response=400, description="Bad request"),
  *      @OA\Response(response=401, description="Unauthorized"),
  *      @OA\Response(response=402, description="Payment Required"),
  *      security={
  *          {"bearerAuth": {}}
  *      }
  * )
  *
  */

/**
  * @OA\Get(
  *      path="/v1/entities/pasien/read?slug=pasien&id={id}",
  *      operationId="readPasien",
  *      tags={"pasien"},
  *      summary="Get Pasien based on id",
  *      description="Returns Pasien based on id",
  *      @OA\Parameter(
  *          name="id",
  *          required=true,
  *          in="path",
  *          @OA\Schema(
  *              type="integer"
  *          )
  *      ),
  *      @OA\Response(response=200, description="Successful operation"),
  *      @OA\Response(response=400, description="Bad request"),
  *      @OA\Response(response=401, description="Unauthorized"),
  *      @OA\Response(response=402, description="Payment Required"),
  *      security={
  *          {"bearerAuth": {}}
  *      }
  * )
  *
  */

/**
  * @OA\Post(
  *      path="/v1/entities/pasien/add",
  *      operationId="addPasien",
  *      tags={"pasien"},
  *      summary="Insert new Pasien",
  *      description="Insert new Pasien into database",
  *      @OA\RequestBody(
  *         @OA\MediaType(
  *             mediaType="application/json",
  *             @OA\Schema(
  *                 @OA\Property(
  *                     property="data",
  *                     type="object",
  *                     example={"id":"123", "ktpId":"", "bpjsId":"", "alergiId":"", "rsId":"", "nama":"Abc", "tmpLahir":"Abc", "tglLahir":"Abc", "alamat":"Abc", "goldar":"Abc", "nik":"123", "jenkel":"Abc", "keterangan":"Abc", "tglCatat":"Abc"},
  *                 ),
  *             )
  *         )
  *      ),
  *      @OA\Response(response=200, description="Successful operation"),
  *      @OA\Response(response=400, description="Bad request"),
  *      @OA\Response(response=401, description="Unauthorized"),
  *      @OA\Response(response=402, description="Payment Required"),
  *      security={
  *          {"bearerAuth": {}}
  *      }
  * )
  *
  */

/**
  * @OA\Put(
  *      path="/v1/entities/pasien/edit",
  *      operationId="editPasien",
  *      tags={"pasien"},
  *      summary="Edit an existing Pasien",
  *      description="Edit an existing Pasien",
  *      @OA\RequestBody(
  *         @OA\MediaType(
  *             mediaType="application/json",
  *             @OA\Schema(
  *                 @OA\Property(
  *                     property="data",
  *                     type="object",
  *                     example={"id":"123", "ktpId":"", "bpjsId":"", "alergiId":"", "rsId":"", "nama":"Abc", "tmpLahir":"Abc", "tglLahir":"Abc", "alamat":"Abc", "goldar":"Abc", "nik":"123", "jenkel":"Abc", "keterangan":"Abc", "tglCatat":"Abc"},
  *                ),
  *             )
  *         )
  *     ),
  *      @OA\Response(response=200, description="Successful operation"),
  *      @OA\Response(response=400, description="Bad request"),
  *      @OA\Response(response=401, description="Unauthorized"),
  *      @OA\Response(response=402, description="Payment Required"),
  *      security={
  *          {"bearerAuth": {}}
  *      }
  * )
  *
  */

/**
  * @OA\Delete(
  *      path="/v1/entities/pasien/delete",
  *      operationId="deletePasien",
  *      tags={"pasien"},
  *      summary="Delete one record of Pasien",
  *      description="Delete one record of Pasien",
  *      @OA\RequestBody(
  *         @OA\MediaType(
  *             mediaType="application/json",
  *             @OA\Schema(
  *                 @OA\Property(
  *                     property="slug",
  *                     example="pasien",
  *                     type="string"
  *                 ),
  *                 @OA\Property(
  *                     property="data",
  *                     type="array",
  *                     @OA\Items(
  *                         type="object",
  *                         @OA\Property(type="string", property="field", default="id"),
  *                         @OA\Property(type="integer", property="value", example="123"),
  *                     ),
  *                ),
  *             )
  *         )
  *     ),
  *      @OA\Response(response=200, description="Successful operation"),
  *      @OA\Response(response=400, description="Bad request"),
  *      @OA\Response(response=401, description="Unauthorized"),
  *      @OA\Response(response=402, description="Payment Required"),
  *      security={
  *          {"bearerAuth": {}}
  *      }
  * )
  *
  */

/**
  * @OA\Delete(
  *      path="/v1/entities/pasien/delete-multiple",
  *      operationId="deleteMultiplePasien",
  *      tags={"pasien"},
  *      summary="Delete multiple record of Pasien",
  *      description="Delete multiple record of Pasien",
  *      @OA\RequestBody(
  *         @OA\MediaType(
  *             mediaType="application/json",
  *             @OA\Schema(
  *                 @OA\Property(
  *                     property="slug",
  *                     example="pasien",
  *                     type="string"
  *                 ),
  *                 @OA\Property(
  *                     property="data",
  *                     type="array",
  *                     @OA\Items(
  *                         type="object",
  *                         @OA\Property(type="string", property="field", default="ids"),
  *                         @OA\Property(type="{}", property="value", example="123,123"),
  *                     ),
  *                ),
  *             )
  *         )
  *     ),
  *      @OA\Response(response=200, description="Successful operation"),
  *      @OA\Response(response=400, description="Bad request"),
  *      @OA\Response(response=401, description="Unauthorized"),
  *      @OA\Response(response=402, description="Payment Required"),
  *      security={
  *          {"bearerAuth": {}}
  *      }
  * )
  *
  */

/**
  * @OA\Put(
  *      path="/v1/entities/pasien/sort",
  *      operationId="sortPasien",
  *      tags={"pasien"},
  *      summary="Sort existing Pasien",
  *      description="Sort existing Pasien",
  *      @OA\RequestBody(
  *         @OA\MediaType(
  *             mediaType="application/json",
  *             @OA\Schema(
  *                 @OA\Property(
  *                     property="slug",
  *                     example="pasien",
  *                     type="string"
  *                 ),
  *                 @OA\Property(
  *                     property="data",
  *                     type="array",
  *                     example={{"id":"123", "ktpId":"", "bpjsId":"", "alergiId":"", "rsId":"", "nama":"Abc", "tmpLahir":"Abc", "tglLahir":"Abc", "alamat":"Abc", "goldar":"Abc", "nik":"123", "createdAt":"2021-01-01T00:00:00.000Z", "updatedAt":"2021-01-01T00:00:00.000Z", "jenkel":"Abc", "keterangan":"Abc", "tglCatat":"Abc"}, {"id":"123", "ktpId":"", "bpjsId":"", "alergiId":"", "rsId":"", "nama":"Abc", "tmpLahir":"Abc", "tglLahir":"Abc", "alamat":"Abc", "goldar":"Abc", "nik":"123", "createdAt":"2021-01-01T00:00:00.000Z", "updatedAt":"2021-01-01T00:00:00.000Z", "jenkel":"Abc", "keterangan":"Abc", "tglCatat":"Abc"}},
  *                     @OA\Items(
  *                         type="object",
  *                         @OA\Property(type="integer", property="id"), 
  *                         @OA\Property(type="string", property="ktpId"), 
  *                         @OA\Property(type="string", property="bpjsId"), 
  *                         @OA\Property(type="string", property="alergiId"), 
  *                         @OA\Property(type="string", property="rsId"), 
  *                         @OA\Property(type="string", property="nama"), 
  *                         @OA\Property(type="string", property="tmpLahir"), 
  *                         @OA\Property(type="string", property="tglLahir"), 
  *                         @OA\Property(type="string", property="alamat"), 
  *                         @OA\Property(type="string", property="goldar"), 
  *                         @OA\Property(type="integer", property="nik"), 
  *                         @OA\Property(type="string", property="createdAt"), 
  *                         @OA\Property(type="string", property="updatedAt"), 
  *                         @OA\Property(type="string", property="jenkel"), 
  *                         @OA\Property(type="string", property="keterangan"), 
  *                         @OA\Property(type="string", property="tglCatat"),
  *                     ),
  *                ),
  *             )
  *         )
  *     ),
  *      @OA\Response(response=200, description="Successful operation"),
  *      @OA\Response(response=400, description="Bad request"),
  *      @OA\Response(response=401, description="Unauthorized"),
  *      @OA\Response(response=402, description="Payment Required"),
  *      security={
  *          {"bearerAuth": {}}
  *      }
  * )
  *
  */

File: mobile/lib/app/data/models/pasien.dart
class PasienModel {
  String? message;
  Data? data;
  String? errors;

  PasienModel({this.message, this.data, this.errors});

  PasienModel.fromJson(Map<String, dynamic> json) {
    message = json['message'];
    data = json['data'] != null ? new Data.fromJson(json['data']) : null;
    errors = json['errors'] != null ? "Error terjadi" : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['message'] = this.message;
    if (this.data != null) {
      data['data'] = this.data!.toJson();
    }
    if (this.errors != null) {
      data['errors'] = this.errors;
    }
    return data;
  }
}

class Data {
  int? id;
  String? nama;
  String? tmpLahir;
  String? tglLahir;
  String? alamat;
  String? goldar;
  String? nik;
  String? ktpId;
  String? bpjsId;
  String? alergiId;
  String? createdAt;
  String? updatedAt;
  String? deletedAt;
  String? rsId;
  String? jenkel;
  String? keterangan;
  String? tglCatat;
  Rs? rs;
  Ktp? ktp;
  Bpjs? bpjs;
  Alergi? alergi;

  Data(
      {this.id,
      this.nama,
      this.tmpLahir,
      this.tglLahir,
      this.alamat,
      this.goldar,
      this.nik,
      this.ktpId,
      this.bpjsId,
      this.alergiId,
      this.createdAt,
      this.updatedAt,
      this.deletedAt,
      this.rsId,
      this.jenkel,
      this.keterangan,
      this.tglCatat,
      this.rs,
      this.ktp,
      this.bpjs,
      this.alergi});

  Data.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    nama = json['nama'];
    tmpLahir = json['tmpLahir'];
    tglLahir = json['tglLahir'];
    alamat = json['alamat'];
    goldar = json['goldar'];
    nik = json['nik'];
    ktpId = json['ktpId'];
    bpjsId = json['bpjsId'];
    alergiId = json['alergiId'];
    createdAt = json['createdAt'];
    updatedAt = json['updatedAt'];
    deletedAt = json['deletedAt'];
    rsId = json['rsId'];
    jenkel = json['jenkel'];
    keterangan = json['keterangan'];
    tglCatat = json['tglCatat'];
    rs = json['rs'] != null ? new Rs.fromJson(json['rs']) : null;
    ktp = json['ktp'] != null ? new Ktp.fromJson(json['ktp']) : null;
    bpjs = json['bpjs'] != null ? new Bpjs.fromJson(json['bpjs']) : null;
    alergi =
        json['alergi'] != null ? new Alergi.fromJson(json['alergi']) : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['nama'] = this.nama;
    data['tmpLahir'] = this.tmpLahir;
    data['tglLahir'] = this.tglLahir;
    data['alamat'] = this.alamat;
    data['goldar'] = this.goldar;
    data['nik'] = this.nik;
    data['ktpId'] = this.ktpId;
    data['bpjsId'] = this.bpjsId;
    data['alergiId'] = this.alergiId;
    data['createdAt'] = this.createdAt;
    data['updatedAt'] = this.updatedAt;
    data['deletedAt'] = this.deletedAt;
    data['rsId'] = this.rsId;
    data['jenkel'] = this.jenkel;
    data['keterangan'] = this.keterangan;
    data['tglCatat'] = this.tglCatat;
    if (this.rs != null) {
      data['rs'] = this.rs!.toJson();
    }
    if (this.ktp != null) {
      data['ktp'] = this.ktp!.toJson();
    }
    if (this.bpjs != null) {
      data['bpjs'] = this.bpjs!.toJson();
    }
    if (this.alergi != null) {
      data['alergi'] = this.alergi!.toJson();
    }
    return data;
  }
}

class Rs {
  int? id;
  String? nama;
  String? alamat;
  String? telepon;
  String? provinsi;
  String? createdAt;
  String? updatedAt;

  Rs(
      {this.id,
      this.nama,
      this.alamat,
      this.telepon,
      this.provinsi,
      this.createdAt,
      this.updatedAt});

  Rs.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    nama = json['nama'];
    alamat = json['alamat'];
    telepon = json['telepon'];
    provinsi = json['provinsi'];
    createdAt = json['createdAt'];
    updatedAt = json['updatedAt'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['nama'] = this.nama;
    data['alamat'] = this.alamat;
    data['telepon'] = this.telepon;
    data['provinsi'] = this.provinsi;
    data['createdAt'] = this.createdAt;
    data['updatedAt'] = this.updatedAt;
    return data;
  }
}

class Ktp {
  int? id;
  String? rfid;
  String? agama;
  String? pekerjaan;
  String? createdAt;
  String? updatedAt;

  Ktp(
      {this.id,
      this.rfid,
      this.agama,
      this.pekerjaan,
      this.createdAt,
      this.updatedAt});

  Ktp.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    rfid = json['rfid'];
    agama = json['agama'];
    pekerjaan = json['pekerjaan'];
    createdAt = json['createdAt'];
    updatedAt = json['updatedAt'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['rfid'] = this.rfid;
    data['agama'] = this.agama;
    data['pekerjaan'] = this.pekerjaan;
    data['createdAt'] = this.createdAt;
    data['updatedAt'] = this.updatedAt;
    return data;
  }
}

class Bpjs {
  int? id;
  String? barcode;
  String? nomor;
  String? faskes;
  String? createdAt;
  String? updatedAt;

  Bpjs(
      {this.id,
      this.barcode,
      this.nomor,
      this.faskes,
      this.createdAt,
      this.updatedAt});

  Bpjs.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    barcode = json['barcode'];
    nomor = json['nomor'];
    faskes = json['faskes'];
    createdAt = json['createdAt'];
    updatedAt = json['updatedAt'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['barcode'] = this.barcode;
    data['nomor'] = this.nomor;
    data['faskes'] = this.faskes;
    data['createdAt'] = this.createdAt;
    data['updatedAt'] = this.updatedAt;
    return data;
  }
}

class Alergi {
  int? id;
  String? data;
  String? createdAt;
  String? updatedAt;

  Alergi({this.id, this.data, this.createdAt, this.updatedAt});

  Alergi.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    data = json['data'];
    createdAt = json['createdAt'];
    updatedAt = json['updatedAt'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['data'] = this.data;
    data['createdAt'] = this.createdAt;
    data['updatedAt'] = this.updatedAt;
    return data;
  }
}